src/t_zset.c

```diff
@@ -1653,6 +1653,34 @@ int zsetZiplistValidateIntegrity(unsigned char *zl, size_t size, int deep) {
     return ret;
 }
 
+/* Return random element from a non empty zset.
+ * sdskey need to be called free.*/
+void zsetTypeRandomElement(robj *zsetobj, unsigned long zsetsize, sds *sdskey, double *score) {
+    if (zsetobj->encoding == OBJ_ENCODING_SKIPLIST) {
+        zset *zs = zsetobj->ptr;
+        dictEntry *de = dictGetFairRandomKey(zs->dict);
+        *sdskey = sdsdup(dictGetKey(de));
+        if (score)
+            *score = *(double*)dictGetVal(de);
+    } else if (zsetobj->encoding == OBJ_ENCODING_ZIPLIST) {
+        char buf[128];
```

Comment:
Is this bound by zset_max_ziplist_value? Some people set this higher, seems like we need to guard against writing past buffer.

Reply:
it's not bound by `zset_max_ziplist_value` it holds the score (a double converted to a string)
so numbers are not expected to be long.
but to be on the safe side, i'll limit the memcpy not to do buffer overrun.

Reply:
Ack, thanks